cmd/analyzer/main.go

```diff
@@ -101,25 +106,32 @@ func analyzer() error {
 			return cmd.FailErr(err, "create docker client")
 		}
 		img, err = local.NewImage(
-			repoName,
+			imageName,
 			dockerClient,
-			local.FromBaseImage(repoName),
+			local.FromBaseImage(imageName),
 		)
 		if err != nil {
 			return cmd.FailErr(err, "access previous image")
 		}
 	} else {
 		img, err = remote.NewImage(
-			repoName,
+			imageName,
 			auth.EnvKeychain(cmd.EnvRegistryAuth),
-			remote.FromBaseImage(repoName),
+			remote.FromBaseImage(imageName),
 		)
 		if err != nil {
 			return cmd.FailErr(err, "access previous image")
 		}
 	}
 
-	md, err := analyzer.Analyze(img)
+	cacheStore, err := cache.MaybeCache(cacheImageTag, cacheDir)
+	if err != nil {
+		return cmd.FailErr(err, "set up cache")
+	}
+	if cacheStore == nil {
+		analyzer.Logger.Warn("Not restoring cached layer metadata, no cache flag specified.")
```

Comment:
Why not log this warning in the `main` function so it is grouped with the other flag validations?

Reply:
Good point, moved.